spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/support/MongoRepositoryFactory.java

```diff
@@ -60,6 +57,7 @@ public class MongoRepositoryFactory extends RepositoryFactorySupport {
 	private final CrudMethodMetadataPostProcessor crudMethodMetadataPostProcessor = new CrudMethodMetadataPostProcessor();
 	private final MongoOperations operations;
 	private final MappingContext<? extends MongoPersistentEntity<?>, MongoPersistentProperty> mappingContext;
+	private MongoRepositoryFragmentsContributor fragmentsContributor = MongoRepositoryFragmentsContributor.DEFAULT;
```

Comment:
Both, the `MongoRepositoryFactory` and `MongoRepositoryFactoryBean` have a reference to `MongoRepositoryFragmentsContributor` and the latter sets the fragments contributor always. This makes me wonder if we should set the default instance in the factory since its always set by the factory bean?

Reply:
`MongoRepositoryFactory` can be used standalone, without a factory bean. Introducing the `FragmentsContributor` concept comes from the need to know ahead of time, which fragments would be contributed. We cannot instantiate during AOT the `MongoRepositoryFactory` as it requires infrastructure (driver) access so we have to fallback to something else (factory bean).

Reply:
Ahh got it. Makes sense to me now. Thanks for the explanation.